src/events/http/createJWTAuthScheme.js

```diff
@@ -0,0 +1,100 @@
+import Boom from '@hapi/boom'
+import jwt from 'jsonwebtoken'
+import serverlessLog from '../../serverlessLog.js'
+
+export default function createAuthScheme(jwtOptions) {
+  const authorizerName = jwtOptions.name
+
+  const identitySourceMatch = /^\$request.header.((?:\w+-?)+\w+)$/.exec(
```

Comment:
Could we get a comment roughly describing what this does for the people who are terrible with regex (like myself)?

Reply:
I struggle with regex too - here's the explanation from [regex 101](https://regex101.com/r/rFteW6/1)
/
^\$request.header.((?:\w+-?)+\w+)$
/
^ asserts position at start of the string
\$ matches the character $ literally (case sensitive)
request matches the characters request literally (case sensitive)
. matches any character (except for line terminators)
header matches the characters header literally (case sensitive)
. matches any character (except for line terminators)
1st Capturing Group ((?:\w+-?)+\w+)
Non-capturing group (?:\w+-?)+
+ Quantifier — Matches between one and unlimited times, as many times as possible, giving back as needed (greedy)
\w+
matches any word character (equal to [a-zA-Z0-9_])
-?
matches the character - literally (case sensitive)
\w+
matches any word character (equal to [a-zA-Z0-9_])
$ asserts position at the end of the string, or before the line terminator right at the end of the string (if any)

Reply:
I suspect it's parsing the authorization header btw @chardos

Reply:
The regex is just copied from src/events/http/createAuthScheme.js 

I did a little digging in the git history. Based on this commit:

https://github.com/dherault/serverless-offline/commit/e52cc2e4087289ba518231a5216d2d83f96ca660

the regex was modified from just grabbing all characters (the `w+` part) to also grabbing hyphens since those are valid characters in a request header.

This doesn't actually parse the auth header. It figures out what the name of the header key is if you have customized the header value that will actually contain the token. That header key is then used when the requests are coming in to grab the auth token from the request.